### PR TITLE
Refactored back/front url handling

### PR DESCRIPTION
**Pull request by Mcray3000**
### New Backend Fetching
Any time you want to make a new backend URL call do it like this one `${import.meta.env.VITE_BACKEND_URL}/entries?is_public=True`

### Env Changes
To do local testing put this in your .env

```
VITE_BACKEND_URL=http://localhost:8000
VITE_FRONTEND_URL=http://localhost:5173
```
Also add a .env to the react frontend folder

### CODE REVIEW

File: packages/express-backend/backend.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ const supabaseKey = process.env.SUPABASE_KEY;
 const supabase = createClient(supabaseUrl, supabaseKey);
 
 const corsOptions = {
-  origin: process.env.BACKEND_URL,
+  origin: "*",
```
Comment (J-Dees) [inline packages/express-backend/backend.js:20]: It is still showing this change despite the rollback commit. Can you double check that it won't be merged in if it doesn't work.

Comment (J-Dees) [inline packages/express-backend/backend.js:20]: It is still showing this change despite the rollback commit. Can you double check that it won't be merged in if it doesn't work.

**Review by J-Dees — APPROVED** [review]

File: packages/express-backend/backend.js (hunk 1 above)
Comment (Mcray3000) [reply packages/express-backend/backend.js]: Yea I rerolled

**Review by TLMatthies — APPROVED** [review]
Looks good if it works

File: packages/react-frontend/src/MyApp.jsx [hunk 2]
```diff
@@ -30,17 +30,17 @@ function MyApp() {
   }
 
   function checkUserLogin(person) {
-    return fetch(`${process.env.BACKEND_URL}/users/login`, {
+    return fetch(`${import.meta.env.VITE_BACKEND_URL}/users/login`, {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
       },
       body: JSON.stringify(person),
```
Comment (TLMatthies) [inline packages/react-frontend/src/MyApp.jsx:38]: This needs to be changed to:
body: JSON.stringify({
        user_name: person.name,
        password: person.password
      }),

**Comment by TLMatthies** [comment]
MainPage.jsx is also missing a replaced process variable at line 25

**Review by eamyzzaj — APPROVED** [review]
Thanks for fixing the frontend/backend connection and also for the detailed description of how to test locally while its deployed! If Tim says it's all good, then I think good to merge.

**Comment by Alaxxn** [comment]
Thanks for getting this figured out.  I was having trouble getting the backend running with the previous changes, so I'm glad we're all back in order. All looks good to me 👍

**Review by Alaxxn — APPROVED** [review]

File: packages/react-frontend/src/MyApp.jsx (hunk 2 above)
Comment (Mcray3000) [reply packages/react-frontend/src/MyApp.jsx:38]: Gotcha, done in Jonathan's branch

**Pull request merged**